torch_xla/csrc/runtime/pjrt_registry.cc

```diff
@@ -21,8 +21,24 @@ namespace runtime {
 
 namespace {
 
+// Placeholder plugin for testing only. Does not implement multiprocessing or
+// configuration. Very likely will not work from Python code.
+class LibraryPlugin : public PjRtPlugin {
+ public:
+  std::string library_path() const override {
+    return sys_util::GetEnvString("PJRT_LIBRARY_PATH", "");
+  }
+
+  const std::unordered_map<std::string, xla::PjRtValueType>
+  client_create_options() const override {
+    return {};
+  }
+
+  bool requires_xla_coordinator() const override { return false; }
+};
```

Comment:
why do we need this plugin?

Reply:
This is just a minimal implementation of the plugin interface that lets us pass in a path. We want to take this codepath during init, rather than the statically linked CUDA client: https://github.com/pytorch/xla/blob/b9a9449f205d10769660c428cc68755a6ffe183a/torch_xla/csrc/runtime/pjrt_registry.cc#L63

 Normally the plugin registration is done through Python, so we need a workaround for the C++ tests.